faw/pdf-observatory/ui/src/views/HomeView.vue

```diff
@@ -1445,16 +1465,24 @@ export default Vue.extend({
           groups[k] = nfiles;
         }
       }
-      
-      let pdfGroups: PdfGroups = {
+
+      const pdfGroups: PdfGroups = {
         groups: groups,
         files: this.pdfGroups.files
       };
-      
-      let newPdfs = reprocessCommon(dd, pdfGroups);
-      this.pdfs = Object.freeze(newPdfs);
-      this.pdfsDslLast = Object.freeze(newPdfs);
-      this.reprocessPost();
+
+      try {
+        const reprocessResult = reprocessCommon(dd, pdfGroups);
+        const newPdfs = reprocessResult.decisions;
+        this.pdfs = Object.freeze(newPdfs);
+        this.pdfsDslLast = Object.freeze(newPdfs);
+        this.extraFeaturesByFile = reprocessResult.extraFeaturesByFile;
```

Comment:
This should be `Object.freeze()` as well, if I'm not mistaken. What's important about Vue here is that, without `Object.freeze()`, it will deeply watch all those strings and whatnot that are part of extraFeaturesByFile. Not a huge issue on the small sets of training data, but definitely an issue on the larger corpus.

Reply:
Thanks, will change.